File: examples/java_example/src/main/java/frc/robot/commands/ShooterRpmCommand.java
package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.subsystems.Shooter;

public class ShooterRpmCommand extends CommandBase {

    private final Shooter m_shooter;
    private final double m_rpm;

    public ShooterRpmCommand(Shooter shooter, double rpm) {
        m_shooter = shooter;
        m_rpm = rpm;

        addRequirements(shooter);
    }

    public void execute() { m_shooter.spinAtRpm(m_rpm); }
    public boolean isFinished() { return false; }
    public void end(boolean interrupted) { m_shooter.stop(); }
}

File: examples/kotlin_example/src/main/kotlin/frc/robot/commands/ShooterRpmCommand.kt
package frc.robot.commands;

import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.subsystems.Shooter;

class ShooterRpmCommand(shooter: Shooter, rpm: Double) : CommandBase() {
    private val m_shooter = shooter;
    private val m_rpm = rpm;

    init {
        addRequirements(m_shooter);
    }

    override fun execute() {
      m_shooter.spinAtRpm(m_rpm);
    }

    override fun isFinished(): Boolean {
      return false;
    }

    override fun end(interrupted: Boolean) {
      m_shooter.stop();
    }
}
